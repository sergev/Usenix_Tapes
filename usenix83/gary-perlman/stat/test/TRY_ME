echo "This is a test of the statistical programs ifrom CSL"
echo "First, let's look at the anova for errors:"
echo "dm s1 s2 s3 s4 s5 < data | anova subject OS time dificulty errors"
dm s1 s2 s3 s4 s5 < data | anova subject OS time dificulty errors
echo "Now let's look at the reaction time data:"
echo "dm s1 s2 s3 s4 s6 < data | anova subject OS time dificulty rt"
dm s1 s2 s3 s4 s6 < data | anova subject OS time dificulty rt
echo "Now let's combine these and look at the efficiency (= #correct/rt):"
echo "dm s1 s2 s3 s4 '(10-x5)/x6' < data | anova subject OS time dificulty efficiency"
dm s1 s2 s3 s4 '(10-x5)/x6' < data | anova subject OS time dificulty efficiency
echo "Let's look at the linear relation between reaction time and errors:"
echo "dm s6 s5 < data | regress rt errors"
dm s6 s5 < data | regress rt errors
echo "We can get similar information from pair as there are only two variables:"
echo "dm s6 s5 < data | pair ps xrt yerrors"
dm s6 s5 < data | pair ps xrt yerrors
echo "We can look at the skew of the distribution or RT's after taking the log:"
echo "dm 'Log(x6)' < data | desc so hi.1m2M3 cfp"
dm 'Log(x6)' < data | desc so hi.1m2M3 cfp
echo "Here's a strange way to get a mean:"
echo "dm x6 < data | transpose | dm \"'MEAN ='\" SUM/N"
dm x6 < data | transpose | dm "'MEAN ='" SUM/N
echo "Hope these examples are useful."
echo "You might want to try out the textbook examples"
echo "to make sure things are working right."
echo "			Gary Perlman"
