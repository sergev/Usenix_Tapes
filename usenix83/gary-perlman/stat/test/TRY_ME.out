This is a test of the statistical programs ifrom CSL
First, let's look at the anova for errors:
dm s1 s2 s3 s4 s5 < data | anova subject OS time dificulty errors
SOURCE: grand mean
OS     time   dific     N       MEAN         SD
                       66     4.3485     2.5267

SOURCE: OS 
OS     time   dific     N       MEAN         SD
UNIX                   36     3.4444     1.8890
UNIQ                   30     5.4333     2.7877

SOURCE: time 
OS     time   dific     N       MEAN         SD
       6hour           33     5.5455     2.2233
       month           33     3.1515     2.2517

SOURCE: OS time 
OS     time   dific     N       MEAN         SD
UNIX   6hour           18     4.5556     1.6881
UNIX   month           18     2.3333     1.3720
UNIQ   6hour           15     6.7333     2.2509
UNIQ   month           15     4.1333     2.7220

SOURCE: dificulty 
OS     time   dific     N       MEAN         SD
              easy     22     2.6364     1.8910
              mediu    22     4.0909     1.9978
              hard     22     6.3182     2.2336

SOURCE: OS dificulty 
OS     time   dific     N       MEAN         SD
UNIX          easy     12     2.1667     1.1146
UNIX          mediu    12     2.9167     1.3114
UNIX          hard     12     5.2500     1.6583
UNIQ          easy     10     3.2000     2.4855
UNIQ          mediu    10     5.5000     1.7795
UNIQ          hard     10     7.6000     2.2211

SOURCE: time dificulty 
OS     time   dific     N       MEAN         SD
       6hour  easy     11     3.9091     1.6404
       6hour  mediu    11     5.0000     1.5492
       6hour  hard     11     7.7273     1.4894
       month  easy     11     1.3636     1.1201
       month  mediu    11     3.1818     2.0405
       month  hard     11     4.9091     1.9725

SOURCE: OS time dificulty 
OS     time   dific     N       MEAN         SD
UNIX   6hour  easy      6     3.0000     0.6325
UNIX   6hour  mediu     6     4.0000     0.6325
UNIX   6hour  hard      6     6.6667     0.5164
UNIX   month  easy      6     1.3333     0.8165
UNIX   month  mediu     6     1.8333     0.7528
UNIX   month  hard      6     3.8333     0.9832
UNIQ   6hour  easy      5     5.0000     1.8708
UNIQ   6hour  mediu     5     6.2000     1.4832
UNIQ   6hour  hard      5     9.0000     1.2247
UNIQ   month  easy      5     1.4000     1.5166
UNIQ   month  mediu     5     4.8000     1.9235
UNIQ   month  hard      5     6.2000     2.1679

FACTOR:    subject         OS       time  dificulty     errors 
LEVELS:         11          2          2          3         66
TYPE  :     RANDOM    BETWEEN     WITHIN     WITHIN       DATA

SOURCE                SS     df             MS         F      p
===============================================================
mean	       1248.0152      1      1248.0152  1235.810  0.000 ***
s/O	          9.0889      9         1.0099

OS     	         64.7293      1        64.7293    64.096  0.000 ***
s/O	          9.0889      9         1.0099

time   	         94.5606      1        94.5606   106.086  0.000 ***
ts/O	          8.0222      9         0.8914

Ot	          0.5838      1         0.5838     0.655  0.439 
ts/O	          8.0222      9         0.8914

dificul	        151.3030      2        75.6515    47.374  0.000 ***
ds/O	         28.7444     18         1.5969

Od	          7.6192      2         3.8096     2.386  0.120 
ds/O	         28.7444     18         1.5969

td	          2.9394      2         1.4697     0.629  0.545 
tds/O	         42.0778     18         2.3377

Otd	          5.3162      2         2.6581     1.137  0.343 
tds/O	         42.0778     18         2.3377

Now let's look at the reaction time data:
dm s1 s2 s3 s4 s6 < data | anova subject OS time dificulty rt
SOURCE: grand mean
OS     time   dific     N       MEAN         SD
                       66   368.0758   236.5611

SOURCE: OS 
OS     time   dific     N       MEAN         SD
UNIX                   36   351.0833   230.7209
UNIQ                   30   388.4667   245.7558

SOURCE: time 
OS     time   dific     N       MEAN         SD
       6hour           33   492.0606   255.1066
       month           33   244.0909   129.9479

SOURCE: OS time 
OS     time   dific     N       MEAN         SD
UNIX   6hour           18   486.5000   251.9387
UNIX   month           18   215.6667    85.3801
UNIQ   6hour           15   498.7333   267.5648
UNIQ   month           15   278.2000   165.7120

SOURCE: dificulty 
OS     time   dific     N       MEAN         SD
              easy     22   260.0455   144.4535
              mediu    22   332.0455   162.2510
              hard     22   512.1364   301.1526

SOURCE: OS dificulty 
OS     time   dific     N       MEAN         SD
UNIX          easy     12   230.1667   115.9646
UNIX          mediu    12   311.5000   132.7311
UNIX          hard     12   511.5833   304.2175
UNIQ          easy     10   295.9000   172.1462
UNIQ          mediu    10   356.7000   196.6091
UNIQ          hard     10   512.8000   313.8491

SOURCE: time dificulty 
OS     time   dific     N       MEAN         SD
       6hour  easy     11   344.6364   157.7988
       6hour  mediu    11   410.1818   182.2453
       6hour  hard     11   721.3636   247.0232
       month  easy     11   175.4545    56.3691
       month  mediu    11   253.9091    92.9424
       month  hard     11   302.9091   182.0090

SOURCE: OS time dificulty 
OS     time   dific     N       MEAN         SD
UNIX   6hour  easy      6   305.8333   119.8589
UNIX   6hour  mediu     6   394.5000   127.7885
UNIX   6hour  hard      6   759.1667   215.8021
UNIX   month  easy      6   154.5000    38.4435
UNIX   month  mediu     6   228.5000    76.7796
UNIX   month  hard      6   264.0000    99.6072
UNIQ   6hour  easy      5   391.2000   198.3046
UNIQ   6hour  mediu     5   429.0000   248.6152
UNIQ   6hour  hard      5   676.0000   299.3693
UNIQ   month  easy      5   200.6000    68.1711
UNIQ   month  mediu     5   284.4000   109.9832
UNIQ   month  hard      5   349.6000   255.7739

FACTOR:    subject         OS       time  dificulty         rt 
LEVELS:         11          2          2          3         66
TYPE  :     RANDOM    BETWEEN     WITHIN     WITHIN       DATA

SOURCE                SS     df             MS         F      p
===============================================================
mean	    8941664.3788      1   8941664.3788   305.581  0.000 ***
s/O	     263350.3833      9     29261.1537

OS     	      22868.4045      1     22868.4045     0.782  0.400 
s/O	     263350.3833      9     29261.1537

time   	    1014568.0152      1   1014568.0152    23.351  0.001 ***
ts/O	     391045.4500      9     43449.4944

Ot	      10350.3682      1     10350.3682     0.238  0.637 
ts/O	     391045.4500      9     43449.4944

dificul	     741888.1212      2    370944.0606    21.074  0.000 ***
ds/O	     316841.3000     18     17602.2944

Od	      11851.9121      2      5925.9561     0.337  0.719 
ds/O	     316841.3000     18     17602.2944

td	     240245.2121      2    120122.6061     3.645  0.047 *
tds/O	     593249.4333     18     32958.3019

Otd	      31216.0212      2     15608.0106     0.474  0.630 
tds/O	     593249.4333     18     32958.3019

Now let's combine these and look at the efficiency (= #correct/rt):
dm s1 s2 s3 s4 '(10-x5)/x6' < data | anova subject OS time dificulty efficiency
SOURCE: grand mean
OS     time   dific     N       MEAN         SD
                       66     0.0245     0.0203

SOURCE: OS 
OS     time   dific     N       MEAN         SD
UNIX                   36     0.0290     0.0207
UNIQ                   30     0.0191     0.0188

SOURCE: time 
OS     time   dific     N       MEAN         SD
       6hour           33     0.0131     0.0108
       month           33     0.0359     0.0213

SOURCE: OS time 
OS     time   dific     N       MEAN         SD
UNIX   6hour           18     0.0162     0.0118
UNIX   month           18     0.0418     0.0198
UNIQ   6hour           15     0.0094     0.0083
UNIQ   month           15     0.0288     0.0214

SOURCE: dificulty 
OS     time   dific     N       MEAN         SD
              easy     22     0.0378     0.0229
              mediu    22     0.0230     0.0164
              hard     22     0.0127     0.0121

SOURCE: OS dificulty 
OS     time   dific     N       MEAN         SD
UNIX          easy     12     0.0426     0.0217
UNIX          mediu    12     0.0292     0.0182
UNIX          hard     12     0.0152     0.0122
UNIQ          easy     10     0.0320     0.0242
UNIQ          mediu    10     0.0157     0.0104
UNIQ          hard     10     0.0096     0.0119

SOURCE: time dificulty 
OS     time   dific     N       MEAN         SD
       6hour  easy     11     0.0210     0.0108
       6hour  mediu    11     0.0146     0.0089
       6hour  hard     11     0.0037     0.0028
       month  easy     11     0.0546     0.0192
       month  mediu    11     0.0314     0.0181
       month  hard     11     0.0217     0.0111

SOURCE: OS time dificulty 
OS     time   dific     N       MEAN         SD
UNIX   6hour  easy      6     0.0259     0.0104
UNIX   6hour  mediu     6     0.0178     0.0097
UNIX   6hour  hard      6     0.0049     0.0021
UNIX   month  easy      6     0.0592     0.0161
UNIX   month  mediu     6     0.0406     0.0180
UNIX   month  hard      6     0.0256     0.0080
UNIQ   6hour  easy      5     0.0151     0.0089
UNIQ   6hour  mediu     5     0.0109     0.0069
UNIQ   6hour  hard      5     0.0023     0.0031
UNIQ   month  easy      5     0.0490     0.0229
UNIQ   month  mediu     5     0.0205     0.0118
UNIQ   month  hard      5     0.0169     0.0132

FACTOR:    subject         OS       time  dificulty efficiency 
LEVELS:         11          2          2          3         66
TYPE  :     RANDOM    BETWEEN     WITHIN     WITHIN       DATA

SOURCE                SS     df             MS         F      p
===============================================================
mean	          0.0396      1         0.0396   203.680  0.000 ***
s/O	          0.0018      9         0.0002

OS     	          0.0016      1         0.0016     8.206  0.019 *
s/O	          0.0018      9         0.0002

time   	          0.0086      1         0.0086    88.075  0.000 ***
ts/O	          0.0009      9         0.0001

Ot	          0.0002      1         0.0002     1.678  0.227 
ts/O	          0.0009      9         0.0001

dificul	          0.0070      2         0.0035    21.074  0.000 ***
ds/O	          0.0030     18         0.0002

Od	          0.0002      2         0.0001     0.526  0.600 
ds/O	          0.0030     18         0.0002

td	          0.0010      2         0.0005     3.427  0.055 
tds/O	          0.0025     18         0.0001

Otd	          0.0001      2         0.0001     0.459  0.639 
tds/O	          0.0025     18         0.0001

Let's look at the linear relation between reaction time and errors:
dm s6 s5 < data | regress rt errors
Analysis for 66 points of 2 variables:
VARIABLE  :         rt     errors 
MIN       :   103.0000     0.0000 
MAX       :   999.0000    10.0000 
MEAN      :   368.0758     4.3485 
SD        :   236.5611     2.5267 
CORRELATION MATRIX:
rt        :     1.0000
errors    :     0.6440     1.0000
VARIABLE  :         rt     errors 
REGRESSION EQUATIONS:
SLOPES    :
rt        :                0.0069 
errors    :    60.2968            
INTERCEPT :   105.8760     1.8165 
SStotal   : 3637474.6212   414.9848 
R-Square  :     0.4148     0.4148 
F(1,64)   :    45.3611    45.3611 
prob (F)  :     0.0000     0.0000 
We can get similar information from pair as there are only two variables:
dm s6 s5 < data | pair ps xrt yerrors
                      rt      errors  Difference
Minimums           0.000       0.000       0.000
Maximums         999.000      10.000     989.000
Sums           24293.000     287.000   24006.000
SumSquares  12579139.000    1663.00012319482.000
Means            368.076       4.348     363.727
SDs              236.561       2.527     234.942
t(65)             12.641      13.981      12.577
p                 -0.000      -0.000      -0.000

 Correlation   r-squared       t(64)           p
       0.644       0.415       6.735       0.000
   Intercept       Slope
       1.816       0.007

      10.000|--------------------------------------------------|
            |                                                 2|
            |                                                  |
            |                      1   1            1          |
            |                                                  |
            |            1 2                                   |
            |                                                  |
            |              1  1 1         1               1  2 |
            |                                                  |
            |        11              1       1       1         |
            |                                                  |
            |       2  1 11  1        1        1               |
       4.348Y                                                  |
            |        1 2 1    1 11  1 2                        |
            |                                                  |
            |          11112 1  21    2                        |
            |                                                  |
            |     1 1121  11                                   |
            |                                                  |
            |      1112                                        |
            |      21                                          |
       0.000|------------------X-------------------------------|
            0.000             368.076                      999.000
We can look at the skew of the distribution or RT's after taking the log:
dm 'Log(x6)' < data | desc so hi.1m2M3 cfp
------------------------------------------------------------
 Under Range    In Range  Over Range         Sum
           0          66           0     164.330
------------------------------------------------------------
        Mean      Median    Midpoint   Geometric    Harmonic
       2.490       2.462       2.506       2.477       2.465
------------------------------------------------------------
          SD   Quart Dev       Range
       0.254       0.196       0.987
------------------------------------------------------------
     Minimum  Quartile 1  Quartile 2  Quartile 3     Maximum
       2.013       2.290       2.462       2.682       3.000
------------------------------------------------------------
        Skew    Kurtosis
       0.346       2.225
------------------------------------------------------------
   Midpt    Freq     Cum    Prop     Cum
   2.050       1       1   0.015   0.015 *
   2.150       8       9   0.121   0.136 ********
   2.250       9      18   0.136   0.273 *********
   2.350       7      25   0.106   0.379 *******
   2.450      12      37   0.182   0.561 ************
   2.550       8      45   0.121   0.682 ********
   2.650       5      50   0.076   0.758 *****
   2.750       7      57   0.106   0.864 *******
   2.850       3      60   0.045   0.909 ***
   2.950       6      66   0.091   1.000 ******
Here's a strange way to get a mean:
dm x6 < data | transpose | dm "'MEAN ='" SUM/N
MEAN =	368.076
Hope these examples are useful.
You might want to try out the textbook examples
to make sure things are working right.
			Gary Perlman
